### PR TITLE
[FLINK-12050][Tests] BlockingShutdownTest fails on Java 9

### PR DESCRIPTION
**Pull request by leesf**
## What is the purpose of the change

Fix error in BlockingShutdownTest on Java 9.

## Brief change log

Add clazz.getName.equals("java.lang.ProcessImpl") in if condition.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by leesf** [comment]
@zentol hi, ProcessImpl has a `pid` field(defined as `private final int pid;`) and it works on Java 9. So should i use pid() method via relection?

**Comment by zentol** [comment]
Interesting, the UNIX version has the field, but the windows version doesn't have it. Please access `pid()` via reflection instead since it this is the "intended" way to get the process ID.

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java [hunk 1]
```diff
@@ -260,6 +260,10 @@ public long getProcessId() {
 				Field pidField = clazz.getDeclaredField("pid");
 				pidField.setAccessible(true);
 				return pidField.getLong(process);
+			} else if (clazz.getName().equals("java.lang.ProcessImpl")) {
+				Method pid = clazz.getDeclaredMethod("pid");
+				pid.setAccessible(true);
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java]: IS this necessary even though the method is public?

Comment (leesf) [reply flink-runtime/src/test/java/org/apache/flink/runtime/testutils/TestJvmProcess.java]: @zentol , yes, if not setAccessible(true), the exception(`cannot access a member of class java.lang.ProcessImpl (in module java.base) with modifiers "public"`) will be thrown.

**Pull request merged**